pkg/configmap/configmap.go

```diff
@@ -0,0 +1,166 @@
+		FilterByTime(configMaps []v1.ConfigMap, olderThan time.Time) (filteredConfigMaps []v1.ConfigMap)
+		FilterByMaxCount(configMaps []v1.ConfigMap, keep int) (filteredConfigMaps []v1.ConfigMap)
+		Print(configMaps []v1.ConfigMap, batch bool)
+	}
+	ConfigMapsService struct {
+		client core.ConfigMapInterface
+		helper kubernetes.Kubernetes
+	}
+)
+
+// Constructor
+func NewConfigMapsService(client core.ConfigMapInterface, helper kubernetes.Kubernetes) ConfigMapsService {
+	return ConfigMapsService{
+		client: client,
+		helper: helper,
+	}
+}
+
+// ListNamesAndLabels return names and labels of Config Maps
+func (cms ConfigMapsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	configMaps, err := cms.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, cm := range configMaps {
+		objectMetas = append(objectMetas, cm.ObjectMeta)
+	}
+	configMapNames, labels := util.GetNamesAndLabels(objectMetas)
+	return configMapNames, labels, nil
+}
+
+// List returns a list of ConfigMaps from a namespace
+func (cms ConfigMapsService) List(listOptions metav1.ListOptions) ([]v1.ConfigMap, error) {
+	configMaps, err := cms.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return configMaps.Items, nil
+}
+
+// GetUnused return unused Config Maps
+func (cms ConfigMapsService) GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error) {
+	var usedConfigMaps []v1.ConfigMap
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(configMaps, func(resource v1.ConfigMap) {
+
+			resourceName := resource.GetName()
+
+			if funk.Contains(usedConfigMaps, resource) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := cms.helper.ResourceContains(namespace, resourceName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedConfigMaps = append(usedConfigMaps, resource)
+			}
+		})
+	})
+
+	for _, resource := range configMaps {
+		if !funk.Contains(usedConfigMaps, resource) {
+			unusedConfigMaps = append(unusedConfigMaps, resource)
+		}
+	}
+
+	return unusedConfigMaps, funcErr
+}
+
+// Delete removes Config Maps
+func (cms ConfigMapsService) Delete(configMaps []v1.ConfigMap) {
+	for _, resource := range configMaps {
+		namespace := resource.GetNamespace()
+		name := resource.GetName()
+
+		log.Infof("Deleting config maps %s/%s", namespace, name)
+
+		err := cms.client.Delete(name, &metav1.DeleteOptions{})
+
+		if err != nil {
+			log.WithError(err).Errorf("Failed to delete config map %s/%s", namespace, name)
+		}
+	}
+}
+
+//FilterByTime returns config maps which are older than specified date
+func (cms ConfigMapsService) FilterByTime(configMaps []v1.ConfigMap, olderThan time.Time) (filteredResources []v1.ConfigMap) {
+	log.WithFields(log.Fields{
+		"olderThan":  olderThan,
+		"configMaps": configMaps,
```

Comment:
When filtering a large list of configmaps, this results in a kilometer-long unreadable log line, as each object is printed with all properties and data. For secrets this may even be a credential leak. It may make more sense to only print the names and time. Or move the log line to line 122 after `filteredResources = append(filteredResources, resource)`, so that we know in debug log that the resource is included in the result set (with just name + time).
Then this log statement may be as simple as `log.WithField("olderThan", olderThan).Debug("Filtering resources older than the specified time")` and then each entry is printed separately later in the log.